components/search/search.tsx

```diff
@@ -0,0 +1,391 @@
+            actions.closeRightHandSide?.();
+            return;
+        }
+        actions.showMentions();
+    };
+
+    const getFlagged = (e: ChangeEvent<HTMLButtonElement>): void => {
+        e.preventDefault();
+        if (props.isFlaggedPosts) {
+            actions.closeRightHandSide?.();
+            return;
+        }
+        actions.showFlaggedPosts();
+    };
+
+    const renderMentionButton = (): JSX.Element => (
+        <HeaderIconWrapper
+            iconComponent={
+                <MentionsIcon
+                    className='icon icon--standard'
+                    aria-hidden='true'
+                />
+            }
+            ariaLabel={true}
+            buttonClass={classNames(
+                'channel-header__icon',
+                {'channel-header__icon--active': props.isMentionSearch},
+            )}
+            buttonId={props.isSideBarRight ? 'sbrChannelHeaderMentionButton' : 'channelHeaderMentionButton'}
+            onClick={searchMentions}
+            tooltipKey={'recentMentions'}
+            isRhsOpen={props.isRhsOpen}
+        />
+    );
+
+    const renderFlagBtn = (): JSX.Element => (
+        <HeaderIconWrapper
+            iconComponent={
+                <FlagIcon className='icon icon--standard'/>
+            }
+            ariaLabel={true}
+            buttonClass={classNames(
+                'channel-header__icon ',
+                {'channel-header__icon--active': props.isFlaggedPosts},
+            )}
+            buttonId={props.isSideBarRight ? 'sbrChannelHeaderFlagButton' : 'channelHeaderFlagButton'}
+            onClick={getFlagged}
+            tooltipKey={'flaggedPosts'}
+            isRhsOpen={props.isRhsOpen}
+        />
+    );
+
+    const renderHintPopover = (): JSX.Element => {
+        let termsUsed = 0;
+
+        searchTerms?.split(/[: ]/g).forEach((word: string): void => {
+            if (searchHintOptions.some(({searchTerm}) => searchTerm.toLowerCase() === word.toLowerCase())) {
+                termsUsed++;
+            }
+        });
+
+        if (visibleSearchHintOptions.length > 0 && !props.isMentionSearch) {
+            const helpClass = `search-help-popover${(focussed && termsUsed <= 2) ? ' visible' : ''}`;
+
+            return (
+                <Popover
+                    id={`${props.isSideBarRight ? 'sbr-' : ''}searchbar-help-popup`}
+                    placement='bottom'
+                    className={helpClass}
+                >
+                    <SearchHint
+                        options={visibleSearchHintOptions}
+                        withTitle={true}
+                        onOptionSelected={handleAddSearchTerm}
+                        onMouseDown={handleSearchHintSelection}
+                        highlightedIndex={highlightedSearchHintIndex}
+                        onOptionHover={setHoverHintIndex}
+                    />
+                </Popover>
+            );
+        }
+
+        return <></>;
```

Comment:
Instead we can do an early exit (this is kind of the practice at Mattermost) - flip the condition and return empty first.